smartswitchplugin/src/main/java/com/kaltura/playkit/plugins/smartswitch/SmartSwitchExecutor.kt

```diff
@@ -0,0 +1,180 @@
+                    inputStream = connection.inputStream
+                    val bufferedReader = BufferedReader(InputStreamReader(inputStream))
+                    val stringBuilder = StringBuilder()
+                    var incomingData: String?
+                    while (bufferedReader.readLine().also { incomingData = it } != null) {
+                        stringBuilder.append(incomingData)
+                    }
+                    bufferedReader.close()
+                    inputStream.close()
+                    log.d("SmartSwitch Response: ${stringBuilder}")
+                    val smartSwitchParser: SmartSwitchParser? = Gson().fromJson(stringBuilder.toString(), SmartSwitchParser::class.java)
+                    if (smartSwitchParser?.smartSwitch != null) {
+                        val cdnList: CDNList? = parseSmartSwitchResponse(smartSwitchParser)
+                        if (cdnList != null) {
+                            resourceUrl = cdnList.URL
+                            log.d("Success response CDN URL: ${cdnList.URL} CDN NAME: ${cdnList.CDN_NAME}")
+                            log.d("Success response CDN CODE: ${cdnList.CDN_CODE} CDN SCORE: ${cdnList.CDN_SCORE}")
+                        } else {
+                            errorMessage = "CDNList is empty"
+                            return error(resourceUrl, errorMessage)
+                        }
+                    } else {
+                        val smartSwitchError: SmartSwitchErrorResponse? = Gson().fromJson(stringBuilder.toString(), SmartSwitchErrorResponse::class.java)
+                        smartSwitchError?.let {
+                            it.messages?.get(0)?.message?.let { message ->
+                                errorMessage = message
+                            }
+                        }
+                        return error(resourceUrl, errorMessage)
+                    }
+                } else {
+                    errorMessage = connection.responseMessage
+                    log.d("connection.responseMessage: $errorMessage")
+                    log.d("connection.responseCode: ${connection.responseCode}")
+                    return error(resourceUrl, errorMessage)
+                }
+            } catch (malformedUrlException: MalformedURLException) {
+                log.d("SmartSwitch MalformedURLException: ${malformedUrlException.message}")
+                malformedUrlException.message?.let {
+                    errorMessage = "SmartSwitch MalformedURLException: $it"
+                }
+                return error(resourceUrl, errorMessage)
+            } catch (exception: IOException) {
+                log.d("SmartSwitch IOException: ${exception.message}")
+                exception.message?.let {
+                    errorMessage = "SmartSwitch IOException: $it"
+                }
+                return error(resourceUrl, errorMessage)
+            } finally {
+                connection?.disconnect()
```

Comment:
@GouravSna 

please  inputStream.close()  in finally 
I think it is also required

Reply:
Done